awx_collection/plugins/modules/tower_workflow_job_template_node.py

```diff
@@ -217,7 +236,13 @@ def main():
     existing_item = module.get_one('workflow_job_template_nodes', **{'data': search_fields})
 
     if state == 'absent':
+        # Look up existing approval node for deletion
+        if existing_item['related'].get('unified_job_template') is not None:
+            existing_approval_node = module.get_endpoint(existing_item['related']['unified_job_template'])['json']
         # If the state was absent we can let the module delete it if needed, the module will handle exiting from this
+        # Delete the Approval Node
+        module.delete_if_needed(existing_approval_node, on_continue=True,)
```

Comment:
I haven't use approval nodes enough to know the mechanics... but if you delete the node the corresponding approval_node won't be auto-deleted?

Reply:
It's hard to tell as you can't get a list of them, will do more research, this was abondence of caution like if you delete a project it doesn't delete a template.

Reply:
Did the testing, this is not needed at all, removing to original